packages/waas/src/secure-store.ts

```diff
@@ -0,0 +1,59 @@
+import { openDB } from 'idb'
+
+export interface SecureStoreBackend {
+    get(dbName: string, dbStoreName: string, key: string): Promise<string | null>
+    set(dbName: string, dbStoreName: string, key: string, value: string): Promise<boolean>
+    delete(dbName: string, dbStoreName: string, key: string): Promise<boolean>
+}
+
+export const getDefaultSecureStoreBackend = (): SecureStoreBackend | null => {
+    if (isIndexedDbAvailable()) {
+        return new IndexedDbSecureStoreBackend()
+    } else {
+        return null
+    }
+}
+
+export function isIndexedDbAvailable(): boolean {
+    return typeof indexedDB === 'object'
+}
+
+export class IndexedDbSecureStoreBackend implements SecureStoreBackend {
+    constructor() {
+        if (!isIndexedDbAvailable()) {
+            throw new Error('IndexedDB is not available')
+        }
+    }
+
+    async get(dbName: string, dbStoreName: string, key: string): Promise<string | null> {
+        const db = await openDB(dbName)
```

Comment:
I don't think we probably want to open and close the db each time we get a value? maybe we can just add "open" and "close" methods to the SecureStoreBackend ..? and we can automatically call .open() in the constructor..? for close(), we can leave it to developers to call if they want. Just some ideas.. lmk what you think

... and, if you dont want to call .open() in the constructor for whatever reason, then add a private instance variable, and open if in the get/set/delete methods if private instance of db is undefined, etc..

Reply:
Done, it now auto-handles db open on first call.